src/manager/network-manager.c

```diff
@@ -241,7 +241,7 @@ int manager_set_link_ipv6_dad(const IfNameIndex *ifidx, int dad) {
         if (r < 0)
                 return r;
 
-        r = set_config_file_int(network, "Network", "IPv6DuplicateAddressDetection", dad);
+        r = set_config_file_bool(network, "Network", "IPv6DuplicateAddressDetection", dad);
         if (r < 0) {
```

Comment:
https://www.freedesktop.org/software/systemd/man/latest/systemd.network.html#IPv6DuplicateAddressDetection= it's a integer